### PR TITLE
Mock: add FOTA target

### PR DESCRIPTION
**Pull request by noonfom**
This commit adds a FOTA target for the FOTA Service mock example.

### CODE REVIEW

File: BLE_GattServer_FOTAService/mock/source/main.cpp [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+            case FOTAService::FOTA_COMMIT:
+            {
+                printf("Committing the update\r\n");
+                disconnect(ble::local_disconnection_reason_t::USER_TERMINATION);
+                break;
+            }
+
+            default:
+            {
+                return static_cast<GattAuthCallbackReply_t>
```
Comment (paul-szczepanek-arm) [inline BLE_GattServer_FOTAService/mock/source/main.cpp:176]: why are you casting a const value? can you not use the value you are casting into direclty?

**Comment by paul-szczepanek-arm** [comment]
Looks like a good candidate to use the new BLEApp class to avoid having the boilerplate code. We need to expose the chainable event handler in the BLEApp through a getter.

File: BLE_GattServer_FOTAService/mock/source/main.cpp (hunk 1 above)
Comment (noonfom) [reply BLE_GattServer_FOTAService/mock/source/main.cpp:176]: Can't implicitly cast `FOTAService::ApplicationError_t` to `GattAuthCallbackReply_t`.

**Comment by noonfom** [comment]
@paul-szczepanek-arm Indeed it is a good candidate for the new BLEApp. But, I think we should add it in a separate commit.

File: BLE_GattServer_FOTAService/mock/source/main.cpp (hunk 1 above)
Comment (paul-szczepanek-arm) [reply BLE_GattServer_FOTAService/mock/source/main.cpp:176]: I see now: /* 0x80 - 0x9F => Application Error */

**Comment by noonfom** [comment]
Closed in favour of #370

**Pull request closed without merging**